pyproject.toml

```diff
@@ -0,0 +1,83 @@
+[build-system]
+requires = ["setuptools>=61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "mango-agents"
+version = "2.1.0"
+authors = [
+  { name="mango Team", email="mango@offis.de" },
+]
+description = "Modular Python Agent Framework"
+readme = "readme.md"
+requires-python = ">=3.10"
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+]
+license = {file="LICENSE"}
+dependencies = [
+    "paho-mqtt>=2.1.0",
+    "python-dateutil>=2.9.0",
+    "dill>=0.3.8",
+    "protobuf>=5.27.2",
+    "networkx>=3.4.1"
+]
+
+[project.optional-dependencies]
+fastjson = [
+    "msgspec>=0.18.6"
+]
+test = [
+    "pytest",
+    "pytest-cov",
+    "pytest-asyncio",
+    "pre-commit"
+]
+
+[project.urls]
+Homepage = "https://mango-agents.readthedocs.io"
+Repository = "https://github.com/OFFIS-DAI/mango"
+Issues = "https://github.com/OFFIS-DAI/mango/issues"
+
+
+[tool.pytest.ini_options]
+asyncio_default_fixture_loop_scope = "function"
+markers = [
+    "mqtt",
+]
+
+[tool.coverage.run]
+omit = ["tests/*"]
+
+[tool.ruff]
+target-version = "py38"
```

Comment:
we should remove the target version here - ruff then takes the one from python-requires (which is 3.10), so we do not have to keep these in sync